### ISSUE TITLE
Oversampling for Multitask Classification

### ISSUE THREAD

**Issue by VIGS25** [deepchem/deepchem#1640]
In the ChemNet fine-tuning (and also the original models trained on MolNet datasets), the authors mention they carry out versampling of the minority class. 

For a single-task classification, this is trivial - you just sample indices of actives, and append them to the original dataset to generate the new dataset, to achieve the desired active / inactive ratio. 

For multi-task classification, how does one go about this? - Being active in one task does not necessarily mean being active in another task, and this can mess up the imbalance when trying to oversample. Also, if we go oversampling task by task, how do we combine the results of the different tasks to get the final dataset?

**Comment by peastman** [comment]
One way of handling cases like this is to do it with weights: give more weight to the underrepresented class on each task.  That's what BalancingTransformer does.  But since you're trying to reproduce a particular paper, you presumably want to do the same thing they did in the paper.  Do they not explain how they handled this?

**Comment by VIGS25** [comment]
Yes, with the Balancing Transformer already present, I was unsure whether to just stick to that or carry out the oversampling technique mentioned. 

In the paper, they unfortunately don't mention how it is done. This is the only text that talks about this.
> The remaining 5/6th or 9/10th of the dataset was then used in the
random 5-fold cross validation approach for training Chemception, and stratification was enforced
for the classification tasks (Tox21, HIV), which meant that the training, validation and test set had
approximately the same ratio of classes. In addition, we noted that the ratio of the classes for
classification tasks were imbalanced, ranging from 1:5 to 1:34. To address this skewed
distribution, we oversampled the minority class; in the construction of the training/validation/test
set, we computed the imbalance ratio and appended additional data from the minority class by that
imbalance ratio. It should be noted that this oversampling step was performed after stratification –
which means that while the same data may be repeated in a particular training/validation/test set,
the data across different training/validation/test sets are unique and do not overlap.

**Comment by peastman** [comment]
That makes it sound like they trained each task independently, rather than training them together as a multitask model.  Maybe write to them to ask?

**Comment by VIGS25** [comment]
In the table they put up regarding the dataset characteristics, they do add multi-task binary classification. 
![image](https://user-images.githubusercontent.com/32259379/61472878-9c53e880-a985-11e9-86cd-1c75c6ef6d89.png)

I have written to them, but they generally reply in a week.

**Comment by VIGS25** [comment]
Closing because results were reproducible using the `dc.trans.BalancingTransformer` as well.
